include/nbl/builtin/hlsl/subgroup/basic_portability.hlsl

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP_BASIC_PORTABILITY_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP_BASIC_PORTABILITY_INCLUDED_
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace subgroup
+{
+	static const uint MinSubgroupSizeLog2 = 2;
+	static const uint MinSubgroupSize = 0x1u << MinSubgroupSizeLog2;
+	
+#ifdef NBL_HLSL_IMPL_GL_NV_shader_thread_group
+	static const uint MaxSubgroupSizeLog2 = 5;
+#elif defined(NBL_HLSL_IMPL_GL_AMD_gcn_shader)||defined(NBL_HLSL_IMPL_GL_ARB_shader_ballot)
+	static const uint MaxSubgroupSizeLog2 = 6;
+#else
+	static const uint MaxSubgroupSizeLog2 = 7;
+#endif
+	static const uint MaxSubgroupSize = (0x1u << MaxSubgroupSizeLog2);
+	
+	// REVIEW: No need for #ifdef check because these are always available in HLSL 2021 it seems
```

Comment:
they are, but their presence in the AST/IR will cause emission of SPIR-V instructions that need extensions enabled for the shader to run

Reply:
nvm everyone with Vk 1.2 supports `basic`